scapy/dadict.py

```diff
@@ -17,12 +19,18 @@
 #  Direct Access dictionary   #
 ###############################
 
+# Used in translate. This wouldn't be required on Python 3,
+# but Python 2's translate does not support the dictionnary handling
+# on strings (it does on unicode... wtf)
+_ALPHABET_NUMBERS = "".join([
+    (x if x in string.ascii_letters or x in "0123456789" else "_")
+    for x in (chr(y) for y in range(256))
+])
 
 def fixname(x):
     if x and str(x[0]) in "0123456789":
         x = "n_" + x
-    return x.translate("________________________________________________0123456789_______ABCDEFGHIJKLMNOPQRSTUVWXYZ______abcdefghijklmnopqrstuvwxyz_____________________________________________________________________________________________________________________________________")  # noqa: E501
-
+    return x.translate(_ALPHABET_NUMBERS)
```

Comment:
I'm sorry but I don't agree with this change. I prefer the previous code (simpler, easir to read and probably faster). If you just want to get rid of the `noqa`, you can split the string, e.g.:
```Python
    return x.translate("________________________________________________"
                       "0123456789_______ABCDEFGHIJKLMNOPQRSTUVWXYZ______"
                       "abcdefghijklmnopqrstuvwxyz____________________________"
                       "______________________________________________________"
                       "___________________________________________________")
```